gcp_variant_transforms/beam_io/vcf_estimate_io.py

```diff
@@ -0,0 +1,219 @@
+    self.samples = samples
+    self.size_in_bytes = size_in_bytes
+
+  def __eq__(self, other):
+    return self.file_name == other.file_name
+
+  def __repr__(self):
+    return 'File Name: {}, Variant Count: {}, Samples: {}, Size: {}'.format(
+        self.file_name,
+        self.estimated_variant_count,
+        self.samples,
+        self.size_in_bytes
+    )
+
+
+class VcfEstimateSource(filebasedsource.FileBasedSource):
+  """A source for reading VCF file estimates."""
+
+  def __init__(self,
+               file_pattern,
+               compression_type=filesystem.CompressionTypes.AUTO,
+               validate=True):
+    # type: (str, str, bool) -> None
+    super(VcfEstimateSource, self).__init__(file_pattern,
+                                            compression_type=compression_type,
+                                            validate=validate,
+                                            splittable=False)
+    self._compression_type = compression_type
+
+  def _get_header_info(self, file_to_read, file_name):
+    """Helper function to extract header data from VCF file"""
+    header_size = 0
+    header_line = file_to_read.readline()
+    # Read and skip all header lines starting with ##. Make sure to calculate
+    # their total size, to marginally better approximate the line count.
+    while (not header_line or
+           not header_line.strip() or header_line.startswith('##')):
+      header_size += len(header_line)
+      header_line = file_to_read.readline()
+    if not header_line.startswith('#'):
+      raise ValueError(('No column-defining header line  was found in file {}.'
+                        .format(file_name)))
+
+    header_size += len(header_line)
+
+    calls = header_line.split()[8:] # Remove #CHROME..INFO mandatory fields.
+    return (header_size,
+            calls if (not calls or calls[0] != 'FORMAT') else calls[1:])
+
+  def _extract_samples(self, header_line):
+    """Removes the common fields from last header line to get sample names."""
+
+
+  def _estimated_variant_count(self, file_to_read, file_name, header_size):
+    """Calculates the approximate number of data lines in the file.
+
+    Extracts the size of the first records data line, and gets the size of the
+    total file size from filesystem. Generates the approximate data line count
+    by subtracting header size from total size and diving it by the single line
+    size.
+    """
+    first_record = file_to_read.readline()
+    while not first_record or not first_record.strip():
+      first_record = file_to_read.readline()
+
+    size_in_bytes = filesystems.FileSystems.match(
+        [file_name])[0].metadata_list[0].size_in_bytes
+    all_lines_size = size_in_bytes
+    if not isinstance(file_to_read, filesystem.CompressedFile):
+      # TODO(#482): Find a better solution to handling compressed files.
+      all_lines_size -= header_size
+    line_size = len(first_record)
+    return float(all_lines_size) / line_size, size_in_bytes
```

Comment:
It feels weird to return two irrelevant variables, especially this is something we cannot tell from the function name or docstring. I don't have a better solution, except that since `size_in_bytes = filesystems.FileSystems.match(
 [file_name])[0].metadata_list[0].size_in_bytes` is not costly, we can calculate again whenever we need it.

Reply:
Done.